azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,258 @@
+            tenantID = process.env["ARM_TENANT_ID"];
+            subscriptionID = process.env["ARM_SUBSCRIPTION_ID"];
+        }
+
+        // If they are still empty, try to get the creds from Az CLI.
+        if (!clientID || !clientSecret || !tenantID || !subscriptionID) {
+            // `create()` will throw an error if the Az CLI is not installed or `az login` has never been run.
+            const cliCredentials = await msRestAzure.AzureCliCredentials.create();
+            subscriptionID = cliCredentials.subscriptionInfo.id;
+            credentials = cliCredentials;
+        } else {
+            credentials = await msRestAzure.loginWithServicePrincipalSecret(clientID, clientSecret, tenantID);
+        }
+
+        return new cdnManagement.CdnManagementClient(credentials, subscriptionID);
+    }
+
+    async check(olds: DynamicProviderInputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.CheckResult> {
+        const cdnClient = await this.getCDNManagementClient();
+
+        // If none of the CDN properties changed, then there is nothing to be validated.
+        if (olds.profileName === news.profileName &&
+            olds.endpointName === news.endpointName &&
+            olds.customDomainHostName === news.customDomainHostName) {
+            return { inputs: news };
+        }
+
+        const validationOutput = await cdnClient.endpoints.validateCustomDomain(
+            news.resourceGroupName,
+            news.profileName,
+            news.endpointName,
+            news.customDomainHostName);
+        if (!validationOutput.customDomainValidated) {
+            return {
+                failures: [{
+                    property: this.name,
+                    reason: validationOutput.reason || "domain_validation_failed"
+                }]
+            };
+        }
+
+        return { inputs: news };
+    }
+
+    async diff(id: string, previousOutput: DynamicProviderOutputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.DiffResult> {
+        const replaces: string[] = [];
+        let changes = false;
+        let deleteBeforeReplace = false;
+
+        if (previousOutput.customDomainHostName !== news.customDomainHostName || previousOutput.name !== this.name) {
+            console.warn("Changing the domain name properties will cause a downtime.")
+
+            deleteBeforeReplace = true;
+            if (previousOutput.customDomainHostName !== news.customDomainHostName) {
+                replaces.push("customDomainHostName");
+            }
+            if (previousOutput.name !== this.name) {
+                replaces.push("customDomainName");
+            }
+        }
+
+        if (previousOutput.endpointName !== news.endpointName) {
+            deleteBeforeReplace = true;
+            replaces.push("endpointName");
+        }
+
+        // HTTPS can be enabled/disabled in-place.
+        if (previousOutput.httpsEnabled !== news.httpsEnabled) {
+            changes = true;
```

Comment:
Changes really ought to be true in all cases that there are changes, even if those changes are already listed in `replaces`.

Reply:
Ah gotcha I'll change it. Reading the doc for [`diff`](https://pulumi.io/reference/programming-model/#diff-id-olds-news) again, I now see what it means, that it is regardless of the type of update; replace vs delete/create.